### PR TITLE
Add recovery schedule

### PR DESCRIPTION
**Pull request by johnwatson484**

### CODE REVIEW

**Review by simonnineteen90 — COMMENTED** [review]
One question and then pending going over the e2e scenarios

File: app/processing/payment/calculate-scheduled-payments/get-schedule.js [hunk 1]
```diff
@@ -1,16 +1,17 @@
+const getExpectedSettlementValue = require('./get-expected-settlement-value')
```
Comment (simonnineteen90) [inline app/processing/payment/calculate-scheduled-payments/get-schedule.js:1]: There are two files named `get-schedule` could consider renaming to more descriptive filenames for ease of understanding. 
`app/processing/components/get-schedule.js`
``app/processing/payment/calculate-scheduled-payments/get-schedule.js

File: app/processing/payment/calculate-scheduled-payments/get-expected-settlement-value.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const getExpectedSettlementValue = (totalValue, totalPayments, segment) => {
```
[the remark this answers is not in the document]
Comment (johnwatson484) [reply app/processing/payment/calculate-scheduled-payments/get-expected-settlement-value.js:1]: Quarterly installment.  The term segment is used throughout the code, so replicated to be consistent.

Doesn't have to be quarterly as the service supports other payment schedules too.

File: app/processing/components/get-schedule.js [hunk 3]
```diff
@@ -8,8 +8,8 @@ const getSchedule = (previousPaymentSchedule, newPaymentSchedule, deltaValue) =>
   const paidSegments = previousPaymentSchedule.filter(x => !x.outstanding)
   newPaymentSchedule.splice(0, paidSegments.length)
   if (deltaValue < 0) {
-    const correctionValue = (Math.abs(deltaValue) / newPaymentSchedule.length) / paidSegments.length
-    newPaymentSchedule.forEach(x => { x.value = x.value - correctionValue })
+    const correctionValue = ((Math.abs(deltaValue) / previousPaymentSchedule.length) * paidSegments.length) / newPaymentSchedule.length
```
[the remark this answers is not in the document]
Comment (johnwatson484) [reply app/processing/components/get-schedule.js:11]: It's avoiding a balloon reduction.  So will work out the amount that should have been reduced from already paid segments and divide by number of remaining segments.

Second line adds that value to each segment so each is reduced evenly.

[the remark this answers is not in the document]
Comment (johnwatson484) [reply app/processing/components/get-schedule.js:11]: Added a comment to explain

File: app/processing/payment/calculate-scheduled-payments/get-schedule.js (hunk 1 above)
Comment (johnwatson484) [reply app/processing/payment/calculate-scheduled-payments/get-schedule.js:1]: Agreed, renamed.

**Pull request merged**